cloud/scope/machine.go

```diff
@@ -103,7 +103,16 @@ func (m *MachineScope) Location() string {
 
 // AvailabilityZone returns the AzureMachine Availability Zone.
 func (m *MachineScope) AvailabilityZone() string {
-	return *m.AzureMachine.Spec.AvailabilityZone.ID
+	// If an explicit AZ is on the azure machine use that
```

Comment:
To best support the failure domain support for KubeadmControlPlane this should probably:
- prioritize the Machine.Spec.FailureDomain field (if set)
- fallback to AzureMachine.Spec.FailureDomain field (if set)
- fallback to AzureMachine.Spec.AvailabilityZone.ID (if set)

The controller should also likely:
- If AzureMachine.Spec.FailureDomain is not set, set it to the value of AzureMachineSpec.AvailabilityZone.ID

This will allow the Cluster API Machine controller to pick up the value and update Machine.Spec.FailureDomain accordingly.

Reply:
Yes that makes sense. I hadn't quite worked out the precedence yet, so this really helps. Thanks.

Reply:
I've added the order define above